### PR TITLE
feat(slugbuilder): run slugbuilder as non-root

### LINKED ISSUE #568: Slugbuilder should not run as root
Postinstall scripts specified in package.json fail to run and cause the warning:

```
npm WARN cannot run in wd myapp@0.0.1 mypostinstallscript (wd=/tmp/build)
```

There's [a lot](https://www.google.com/search?q=warn+cannot+run+in+wd) written about this error and it seems to be the result of building as root.

### PR DESCRIPTION
**Pull request by bacongobbler**
This is a security fix as well as a fix for buildpacks that cannot
be run as root.

One small fix had to be made here with openssl. Docker by default
hardcodes $HOME to /, so shifting the export HOME line up before we
call openssl gives openssl write access to $HOME/.rnd.

fixes deis/deis#568

### CODE REVIEW

**Comment by titanous** [comment]
GitHub claims it can't merge this cleanly, would you rebase?

**Comment by bacongobbler** [comment]
:tada: :)

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,14 @@
 FROM progrium/cedarish
 MAINTAINER progrium "progrium@gmail.com"
 
+# add a slugbuilder user
+RUN useradd slugbuilder --shell /bin/bash --home-dir /app
```
Comment (bacongobbler) [inline Dockerfile]: It seems like a Docker bug with `--home-dir` not being respected for the `slugbuilder` user, but we already export HOME to /app in `builder/build.sh`, so it's okay. `--shell` isn't even required, nor necessary. Let me know if you want me to remove these.

Comment (titanous) [inline Dockerfile]: I think we should leave `--home-dir`, and drop `--shell`.

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,14 @@
 FROM progrium/cedarish
 MAINTAINER progrium "progrium@gmail.com"
 
+# add a slugbuilder user
```
Comment (titanous) [inline Dockerfile]: This comment doesn't add any information.

**Comment by titanous** [comment]
Thanks. I'll test with Flynn in a bit and then merge.

**Pull request merged**